### DISCUSSION TITLE
Is 704x480 too low for object detection?

### DISCUSSION THREAD

**Discussion by ronisaacson** [blakeblackshear/frigate#6127] [Ask A Question] [answered]
I know this is a subjective question... but my Amcrest cameras top out at 704x480 on the substream. They're all installed outside (driveway, yard, etc). I have a Coral USB and I know the input is cropped to 320x320 for object detection, but I'm concerned that 704x480 is too low resolution to accurately detect smaller objects, e.g. a person at the far end of the driveway. Currently I'm using the main stream at 1280x720 for both detection and recording, but I'd really like to increase the recording resolution.

Does anyone have experience (good or bad) doing detection with a resolution this low?

**Comment by NickM-27** [comment, marked as answer]
It'll definitely be less accurate at farther distances. If you have a GPU and you're on 0.12, you could use the higher resolution for recording and then use that same stream for detect but set the detect width / height lower which will use the GPU to resize to that resolution. Essentially creating a new sub stream.

**Reply by ronisaacson** [reply]
Ok thanks, I saw that option but I'm concerned about performance with resizing 5 feeds. I guess I'll have to try it out and see!

**Comment by vajonam** [comment]
Is the main reason for the sub stream slower FPS or lower resolution? if you do the above resizing the stream to the detector the FPS still remains high? 

is the Inference Speed a combination of both res and fps, where is really res * fps ~ infrence speed, where you can increase one and decrease the other and stay with similar inference speeds?

**Reply by NickM-27** [reply]
You want both a lower res stream and low fps, mainly because higher resolution and higher fps require more processing without much benefit. 

Inference speed is not affected by resolution since all regions are cropped before sending to the detector anyway (the detector only supports one size image, 320x320 by default)

Inference speed is also not affected by fps, just if your inference speed is high it won't be able to handle high fps.

**Comment by vajonam** [comment]
so then the open.cv2 motion detection is what is liming factoring with regards to fps and resolution?

**Reply by NickM-27** [reply]
Motion detection uses more CPU when the resolution and fps are higher, yes.

**Comment by vajonam** [comment]
just thinking out loud is there an option to take the low res/low fps for motion detection but then crop the image from higher res stream and send that to detector? assuming the record stream are similar aspect ratios, I guess the tricky part is lining up the exact frames since i-frames and frames will not match up..

**Reply by NickM-27** [reply]
You'd have to decode both streams and at the end of the day it doesn't matter what stream you take the image from it is still being cropped down in size so I see no advantage. That's why even if you have beefy hardware there's no reason to run detect on more than 720p in most cases.

**Reply by vajonam** [reply]
this is OpenCV with cuda support, does the existing -tensorrt image already use this? I am using TensorRT (with nvidia GPUs, I have 2 in the this box specifically for NVR stuff).

**Reply by NickM-27** [reply]
No, both images use opencv on the CPU.

**Reply by vajonam** [reply]
is this being worked on or a roadmap? if not if its just a question of packaging and rebuilding images with the right packages I can give it a try. unless you think the benefit of using a `record` quality 30fs HD,4K stream is minimal and I shouldn't bother..

**Reply by NickM-27** [reply]
Always open to contributions but I don't think it's on the main roadmap. And I want to be clear, there are basically no cases where detect would benefit from running on 4k 30fps even if resource usage is not a factor. It is not recommended because it just won't produce better results.

**Reply by vajonam** [reply]
that is what I wanted to understand. thanks for making it clear.
